design/ceph/object/store.md

```diff
@@ -351,3 +351,212 @@ spec:
 
 When this feature is enabled the endpoint in OBCs and COSI Bucket Access can be clubbed with bucket name and host name. For OBC  the `BUCKET_NAME` and the `BUCKET_HOST` from the config map combine to `http://$BUCKETNAME.$BUCKETHOST`. For COSI Bucket Access the `bucketName` and the `endpoint` in the `BucketInfo` to `http://bucketName.endpoint`.
+
+### Pool Placement targets and Storage Classes
+Object Storage API allows users to specify where bucket data will be stored during bucket creation. With `<LocationConstraint>` parameter in S3 API and `X-Storage-Policy` header in SWIFT. Similarly, users can override where object data will be stored by setting `X-Amz-Storage-Class` and `X-Object-Storage-Class` during object creation. 
+
+In Ceph these data locations represented by Pools. Placement targets control which Pools are associated with a particular bucket. A bucket’s placement target is selected on creation, and cannot be modified (See [Ceph doc](https://docs.ceph.com/en/latest/radosgw/placement/#pool-placement-and-storage-classes)). Placement target can also define custom Storage classes to override data pool where object will be stored.
+
+Ceph administrator creates set of `index_pool`, `data_pool`, and `data_extra_pool` per each target placement. Select arbitrary names for target placements. For example `hot` and `cold`:
+```yaml
+# Bucket Data pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+---
+# Bucket index pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket data extra pool for hot storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: hot.rgw.buckets.non-ec
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket Data pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+---
+# Bucket index pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.index
+  namespace: rook-ceph
+spec:
+  ...
+---
+# Bucket data extra pool for cold storage
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: cold.rgw.buckets.non-ec
+  namespace: rook-ceph 
+spec:
+  ...
+```
+If needed, create additional pool to use it in custom StorageClass, for example `GLACIER`:
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: glacier.rgw.buckets.data
+  namespace: rook-ceph 
+spec:
+  ...
+```
+Then update `CephObjectStore` CRD with target placements and StorageClasses definitions:
+```yaml
+spec:
+  targetPlacements:                         # [optional] list of target placements
```

Comment:
I had a chat with Travis. We'd like to follow the trend that Travis used here: https://github.com/rook/rook/pull/13849

We do want to add these options to the CephObjectZone CRD, even though we aren't officially supporting or testing multisite as a first target. This will ensure that when the design is implemented, the code writer is encouraged to write reusable configuration methods that will allow for multisite support in the future. 

These reusable methods are important because it can be exceptionally difficult to convert a single-site-only feature to multisite in the future if individual bits of code/API need to be extracted and combined. 

Overall, I don't expect that this should affect the written code in a big way, either in time taken or number of lines written. Instead, it encourages good coding practices and means that multisite support in the future doesn't require complex code manipulation.

We can note in the Zone CRD that the section is experimental, and that'll be fine.

Reply:
sorry, it was too much information for me in #13849. Can you please be a little be more specific about which trend should be followed? Should we replace placement `[]TargetPlacement` with `map[string]TargetPlacement`?